bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -175,6 +179,58 @@ public void loadConf(CompositeConfiguration baseConf) {
         }
     }
 
+    /**
+     * Get metadata service uri.
+     *
+     * @return metadata service uri.
+     */
+    @SuppressWarnings("deprecation")
+    public String getMetadataServiceUri() throws ConfigurationException {
+        String serviceUri = getString(METADATA_SERVICE_URI);
+        if (null == serviceUri) {
+            // no service uri is defined, fallback to old settings
+            String ledgerManagerType;
+            Class<? extends LedgerManagerFactory> factoryClass = getLedgerManagerFactoryClass();
+            if (factoryClass == null) {
+                ledgerManagerType = HierarchicalLedgerManagerFactory.NAME;
+            } else {
+                if (!AbstractZkLedgerManagerFactory.class.isAssignableFrom(factoryClass)) {
+                    // this is a non-zk implementation
+                    throw new UnsupportedOperationException("metadata service uri is not supported for "
```

Comment:
This is a no service URI case. But the log messages reads so..

Reply:
the idea was: in the case no service uri is defined, if there is a customized factory class implemented, we should fail it and force the users to define a metadata service uri for this factory implementation. so from this perspective, it is a unsupported operation exception.

but if you have a better idea on what exception we should throw here, please help suggest one.